### PR TITLE
Avoid dots in config doc ids as it's causing issues for downstream

### PR DESCRIPTION
**Pull request by gsmet**
I think I went through all the links pointing to generated config doc...

/cc @quarkusio/documentation 
/cc @yrodiere you might want to check I didn't break anything in your doc, even if I tried to be thorough and use cross references to enable the checks

### CODE REVIEW

**Comment by gsmet** [comment]
@MichalMaler for now, I marked it for 3.7 only but I could be convinced to backport it to 3.2 if needed.

**Review by MichalMaler — APPROVED** [review]

**Comment by MichalMaler** [comment]
@gsmet Hey! This is awesome!
Thank you so much!
Will test it on Monday. Yes, we will need this on 3.2, just for our testing and for others to get familiar with the PV1 publishing process and debugging.

File: core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/DocFormatter.java [hunk 1]
```diff
@@ -53,8 +53,8 @@ default String getAnchor(String string) {
 
         // Apostrophes.
         string = string.replaceAll("([a-z])'s([^a-z])", "$1s$2");
-        // Allow only letters, -, _, .
-        string = string.replaceAll("[^\\w-_\\.]", "-").replaceAll("-{2,}", "-");
+        // Allow only letters, -, _
+        string = string.replaceAll("[^\\w-_]", "-").replaceAll("-{2,}", "-");
```
Comment (yrodiere) [inline core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/DocFormatter.java:57]: So we're breaking every single link pointing to Quarkus configuration properties from other websites? I left quite a few of those out there, on stackoverflow, Zulip, Discourse, GitHub, ...

Can we add a secondary anchor for the deprecated, dot-containing id?

File: docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc [hunk 2]
```diff
@@ -948,7 +948,7 @@ To enable the `outbox-polling` coordination strategy, an additional extension is
 include::{includes}/devtools/extension-add.adoc[]
 
 Once the extension is there, you will need to explicitly select the `outbox-polling` strategy
-by setting link:#quarkus-hibernate-search-orm-elasticsearch_quarkus.hibernate-search-orm.coordination.strategy[`quarkus.hibernate-search-orm.coordination.strategy`]
+by setting <<quarkus-hibernate-search-orm-elasticsearch_quarkus.hibernate-search-orm.coordination.strategy,`quarkus.hibernate-search-orm.coordination.strategy`>>
```
Comment (yrodiere) [inline docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc:951]:
```suggestion
by setting <<quarkus-hibernate-search-orm-elasticsearch_quarkus-hibernate-search-orm-coordination-strategy,`quarkus.hibernate-search-orm.coordination.strategy`>>
```

File: docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc [hunk 3]
```diff
@@ -992,7 +992,7 @@ However, there is one key difference: index updates are necessarily asynchronous
 they are guaranteed to happen _eventually_, but not immediately.
 
 This means in particular that the configuration property
-link:#quarkus-hibernate-search-orm-elasticsearch_quarkus.hibernate-search-orm.indexing.plan.synchronization.strategy[`quarkus.hibernate-search-orm.indexing.plan.synchronization.strategy`]
+<<quarkus-hibernate-search-orm-elasticsearch_quarkus.hibernate-search-orm.indexing.plan.synchronization.strategy,`quarkus.hibernate-search-orm.indexing.plan.synchronization.strategy`>>
```
Comment (yrodiere) [inline docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc:995]:
```suggestion
<<quarkus-hibernate-search-orm-elasticsearch_quarkus-hibernate-search-orm-indexing-plan-synchronization-strategy,`quarkus.hibernate-search-orm.indexing.plan.synchronization.strategy`>>
```

File: docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc [hunk 4]
```diff
@@ -965,16 +965,16 @@ link:{hibernate-search-docs-url}#coordination-outbox-polling-schema[manually alt
...
+* <<quarkus-hibernate-search-orm-outboxpolling_quarkus.hibernate-search-orm-coordination-entity-mapping-outbox-event-catalog,`quarkus.hibernate-search-orm.coordination.entity-mapping.outbox-event.catalog`>>
+* <<quarkus-hibernate-search-orm-outboxpolling_quarkus.hibernate-search-orm-coordination-entity-mapping-outbox-event-schema,`quarkus.hibernate-search-orm.coordination.entity-mapping.outbox-event.schema`>>
+* <<quarkus-hibernate-search-orm-outboxpolling_quarkus.hibernate-search-orm-coordination-entity-mapping-outbox-event-table,`quarkus.hibernate-search-orm.coordination.entity-mapping.outbox-event.table`>>
+* <<quarkus-hibernate-search-orm-outboxpolling_quarkus.hibernate-search-orm-coordination-entity-mapping-outbox-event-uuid-gen-strategy,`quarkus.hibernate-search-orm.coordination.entity-mapping.outbox-event.uuid-gen-strategy`>>
+* <<quarkus-hibernate-search-orm-outboxpolling_quarkus.hibernate-search-orm-coordination-entity-mapping-outbox-event-uuid-type,`quarkus.hibernate-search-orm.coordination.entity-mapping.outbox-event.uuid-type`>>
```
Comment (yrodiere) [inline docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc:977]: This looks fine, but I couldn't check it because of #38443

**Review by yrodiere — CHANGES_REQUESTED** [review]
LGTM apart from a few oversights, see suggestions below.

Also, while I'm absolutely not blocking this PR for that reason, I don't like that pretty much all links to the Quarkus documentation that I've posted on the web so far will get broken because of this. IMO whatever bug in downstream tools led to this PR should be fixed in downstream tools instead.

File: docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc (hunk 4 above)
Comment (yrodiere) [reply docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc:977]: Well actually I could check it, dumb me. Please ignore this.

Comment (yrodiere) [inline docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc:977]:
```suggestion
* <<quarkus-hibernate-search-orm-outboxpolling_quarkus-hibernate-search-orm-coordination-entity-mapping-agent-catalog,`quarkus.hibernate-search-orm.coordination.entity-mapping.agent.catalog`>>
* <<quarkus-hibernate-search-orm-outboxpolling_quarkus-hibernate-search-orm-coordination-entity-mapping-agent-schema,`quarkus.hibernate-search-orm.coordination.entity-mapping.agent.schema`>>
* <<quarkus-hibernate-search-orm-outboxpolling_quarkus-hibernate-search-orm-coordination-entity-mapping-agent-table,`quarkus.hibernate-search-orm.coordination.entity-mapping.agent.table`>>
* <<quarkus-hibernate-search-orm-outboxpolling_quarkus-hibernate-search-orm-coordination-entity-mapping-agent-uuid-gen-strategy,`quarkus.hibernate-search-orm.coordination.entity-mapping.agent.uuid-gen-strategy`>>
* <<quarkus-hibernate-search-orm-outboxpolling_quarkus-hibernate-search-orm-coordination-entity-mapping-agent-uuid-type,`quarkus.hibernate-search-orm.coordination.entity-mapping.agent.uuid-type`>>
* <<quarkus-hibernate-search-orm-outboxpolling_quarkus-hibernate-search-orm-coordination-entity-mapping-outbox-event-catalog,`quarkus.hibernate-search-orm.coordination.entity-mapping.outbox-event.catalog`>>
* <<quarkus-hibernate-search-orm-outboxpolling_quarkus-hibernate-search-orm-coordination-entity-mapping-outbox-event-schema,`quarkus.hibernate-search-orm.coordination.entity-mapping.outbox-event.schema`>>
* <<quarkus-hibernate-search-orm-outboxpolling_quarkus-hibernate-search-orm-coordination-entity-mapping-outbox-event-table,`quarkus.hibernate-search-orm.coordination.entity-mapping.outbox-event.table`>>
* <<quarkus-hibernate-search-orm-outboxpolling_quarkus-hibernate-search-orm-coordination-entity-mapping-outbox-event-uuid-gen-strategy,`quarkus.hibernate-search-orm.coordination.entity-mapping.outbox-event.uuid-gen-strategy`>>
* <<quarkus-hibernate-search-orm-outboxpolling_quarkus-hibernate-search-orm-coordination-entity-mapping-outbox-event-uuid-type,`quarkus.hibernate-search-orm.coordination.entity-mapping.outbox-event.uuid-type`>>
```

**Review by yrodiere — CHANGES_REQUESTED** [review]

**Comment by MichalMaler** [comment]
@yrodiere  Hello Yoann,

I think you at least deserve to know why this is happening. RHBQ was stuck for a long time on the old and unsupported publishing platform Pantheon II. We implemented a few workarounds to make publishing possible, yet operating on that tool was painful. Also, we were the last team there; the others had already fled to something else.

While the migration to the older, but technically savvier brother, Pantheon 1, offers many benefits and solves hundreds of small issues, it causes the one we are resolving right now.

I asked Guillaume if the transformation of these links and IDs could be part of our script, but he proposed this approach and unification. I am sorry to hear that it causes some problems, but once this is resolved, we should be in a much better position and operate on an overall better publishing tool.

On behalf of the entire docs team, I want to apologize for any additional workload and problem-solving these changes brings.

**Comment by yrodiere** [comment]
I understand, Michal. Thanks for the details.

I'd simply suggest that moving forward, the docs team considers the impact of breaking links. And I don't mean just in this specific case: I suspect we've had several similar changes in the past, like renaming a guide, or simply changing headings that relied on [auto-generated anchors](https://docs.asciidoctor.org/asciidoc/latest/macros/xref/#anchors).

It's not that it causes additional work for upstream, it's more that it drastically reduces the amount of time that that work will remain useful. We have lots of discussions and questions/answers referencing upstream docs out there, and every time we make a change like this, we make a few of those discussions obsolete, by preventing people to follow relevant links. That's basically getting rid of part of the "documentation ecosystem", one piece at a time.

I understand we can't keep compatibility forever, but I think that problem needs to at least be considered, and mitigated where possible (which is not here, from what I understand).

**Comment by MichalMaler** [comment]
@yrodiere 
The changes we need can be part of the script we are already using, so that Upstream content remains the same, and RHBQ repo will get what it needs. I am just not good in programming to propose something that could be applicable.
I sent you an email with our  bccutil's problem description, just so you have all the info you would need to make a call.

**Comment by MichalMaler** [comment]
@gsmet Hello Guillaume. That 3.2 backport would be nice, but it is not mandatory.
If it will not be a fast forward merge that would required some additional work, please drop the initiative. Dont want to require your time on something that is not 100% needed, as this ID transformation.
If it is easy to do, I will be glad. 

Also, as I wrote in our email conversation, we will be good with both solutions:
1. Applying this transformation only on our RHBQ set of filles that will be applied during the single-sourcing.
2. Or direct patch as you are proposing here.

Much thanks.

**Comment by gsmet** [comment]
> The changes we need can be part of the script we are already using, so that Upstream content remains the same

Unfortunately, it's not easy to identify these links in a reliable way, thus why I did the change for upstream too.

It's something we will have to live with.

File: docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc [hunk 5]
```diff
@@ -898,10 +898,10 @@ this could cause a startup failure.
 To address this, you can configure Hibernate Search to not send any request on startup:
 
 * Disable Elasticsearch version checks on startup by setting the configuration property
-  link:#quarkus-hibernate-search-orm-elasticsearch_quarkus.hibernate-search-orm.elasticsearch.version-check.enabled[`quarkus.hibernate-search-orm.elasticsearch.version-check.enabled`]
+  <<quarkus-hibernate-search-orm-elasticsearch_quarkus-hibernate-search-orm-elasticsearch-version-check.enabled,`quarkus.hibernate-search-orm.elasticsearch.version-check.enabled`>>
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc:901]: @gsmet looks like you erased my commit, so I opened a separate PR: https://github.com/quarkusio/quarkus/pull/38459

You might still want to apply this:

```suggestion
  <<quarkus-hibernate-search-orm-elasticsearch_quarkus-hibernate-search-orm-elasticsearch-version-check-enabled,`quarkus.hibernate-search-orm.elasticsearch.version-check.enabled`>>
```

**Review by yrodiere — APPROVED** [review]

**Comment by ppalaga** [comment]
Ouch, this hits also Camel Quarkus and Quarkus CXF. We can invest the work in our current branches, but pretty please do not backport this to 3.2.

**Pull request merged**